### PR TITLE
fail earlier when force=FALSE?

### LINKED ISSUE #167: is this a bug
``` r
library("fledge")
with_demo_project(quiet = TRUE, {
    usethis::use_r("bla")
    gert::git_add("R/bla.R")
    gert::git_commit("* Add cool bla.")
    tag_version()
    get_last_tag()[, c("name", "ref")]

    tag_version(force = FALSE)

})
#> 
#> ── Tagging Version ──
#> 
#> → Creating tag v0.0.0.9000 with tag message derived from 'NEWS.md'.
#> 
#> ── Tagging Version ──
#> 
#> ℹ Tag v0.0.0.9000 exists and points to the current commit.
#> → Creating tag v0.0.0.9000 with tag message derived from 'NEWS.md'.
#> Error in libgit2::git_tag_create: tag already exists
```

<sup>Created on 2021-12-02 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

### PR DESCRIPTION
**Pull request by maelle**
Fix #167 

@krlmlr I might be misunderstanding something but if the tag already exists, the gert call is going to fail, no matter the commit the existing tag points to?

### CODE REVIEW

**Comment by krlmlr** [comment]
I just didn't bother to add a proper error message for this cas, knowing that gert will fail.

**Comment by maelle** [comment]
But there _is_ a message, it just wasn't at the ideal place in the if-branches? Are the tweaks I made incorrect?

**Comment by krlmlr** [comment]
The tag should be deleted only if `force = TRUE` .

**Comment by krlmlr** [comment]
Oh, I see now. Maybe we should silently `return()` if the tag exists and points to the same commit?

**Comment by maelle** [comment]
> Oh, I see now. Maybe we should silently return() if the tag exists and points to the same commit?

Awesome, I'll do that.

**Comment by maelle** [comment]
@krlmlr here's the new version


```r
tag_version_impl <- function(force) {
  check_only_staged(character())

  current_news <- get_current_news()
  desc <- desc::desc(file = "DESCRIPTION")
  version <- desc$get_version()

  if (fledge_chatty()) cli_h2("Tagging Version")

  tag <- paste0("v", version)

  if (tag_already_exist(tag)) {

    # Tagging would do nothing as the tag points to the current commit
    if (gert::git_commit_info(tag)$id == gert::git_log(max = 1)$commit) {
      if (fledge_chatty()) {
        cli_alert_info("Tag {.field {tag}} exists and points to the current commit.")
      }
      return()
    }

    # Tagging would fail
    if (!force) {
      abort(paste0("Tag ", tag, " exists, use `force = TRUE` to overwrite."))
    }

    if (fledge_chatty()) cli_alert("Deleting existing tag {.field {tag}}.")
    gert::git_tag_delete(tag)
  }

  if (fledge_chatty()) {
    cli_alert(
      "Creating tag {.field {tag}} with tag message derived from {.file NEWS.md}.",
      wrap = TRUE
    )
  }

  msg_header <- paste0(desc$get("Package"), " ", version)
  gert::git_tag_create(tag, message = paste0(msg_header, "\n\n", current_news))

  invisible(tag)
}


```

File: R/tag-version.R [hunk 1]
```diff
@@ -53,3 +59,7 @@ get_current_news <- function() {
   current_news <- paste(current_news, collapse = "\n")
   gsub("^\n*(.*[^\n])\n*$", "\\1", current_news)
 }
+
+tag_already_exist <- function(tag) {
```
Comment (maelle) [inline R/tag-version.R:63]: this was for readability

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**